File: freeq-android/freeq/src/main/java/com/freeq/model/ServerConfig.kt
package com.freeq.model

/** Central server configuration - change here to point to a different server */
object ServerConfig {
    /** IRC server host:port (default: production) */
    var ircServer: String = "irc.freeq.at:6667"

    /** HTTPS API base URL (derived from ircServer) */
    val apiBaseUrl: String
        get() = "https://" + ircServer.substringBefore(":")

    /** Auth broker base URL (default: production standalone broker) */
    // For deployments using embedded auth (no standalone broker), use apiBaseUrl:
    // val authBrokerBase: String get() = apiBaseUrl
    val authBrokerBase: String = "https://auth.freeq.at"
}

File: freeq-ios/freeq/Models/ServerConfig.swift
import Foundation

/// Central server configuration - change here to point to a different server
struct ServerConfig {
    /// IRC server host:port — used as the TCP fallback when WebSocket can't
    /// be reached. Many cellular / captive-portal / corporate networks block
    /// raw 6667, which is why we prefer the WebSocket URL below.
    static var ircServer: String = "irc.freeq.at:6667"

    /// Primary IRC transport — same WebSocket endpoint the web client uses.
    /// Lives on port 443, so it traverses every firewall that allows HTTPS.
    static var wssServer: String = "wss://irc.freeq.at/irc"

    /// HTTPS API base URL (derived from ircServer)
    static var apiBaseUrl: String {
        let host = ircServer.components(separatedBy: ":").first ?? ircServer
        return "https://\(host)"
    }
}
